aeron-driver/src/main/java/io/aeron/driver/Configuration.java

```diff
@@ -1187,4 +1186,76 @@ public static long parseSize(final String propertyName, final String propertyVal
                     "Trailing character should be one of k, m, or g, but was " + lastCharacter);
         }
     }
+
+    private static long getTimeInNanos(final String propertyName, final long defaultValue)
+    {
+        final String propertyValue = getProperty(propertyName);
+        if (propertyValue != null)
+        {
+            final long value = parseTime(propertyName, propertyValue);
+            if (value < 0)
+            {
+                throw new ConfigurationException(
+                        "Value " + value + " for property " + propertyName + " must be positive");
+            }
+
+            if (value == Long.MAX_VALUE)
```

Comment:
This bit is here rather than below since all of the `TimeUnits...toNanos` calls return `Long.MAX_VALUE` when they hit positive overflow. I feel like not allowing `Long.MAX_VALUE` is an ok trade-off?